lib/pf/UnifiedApi/Controller/Redis.pm

```diff
@@ -0,0 +1,78 @@
+package pf::UnifiedApi::Controller::Redis;
+
+=head1 NAME
+
+pf::UnifiedApi::Controller::Redis -
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::UnifiedApi::Controller::Redis
+
+=cut
+
+use strict;
+use warnings;
+use Mojo::Base 'pf::UnifiedApi::Controller::RestRoute';
+use pf::pfqueue::stats;
+
+
+sub queue {
+    my ($self) = @_;
+    my $stats = pf::pfqueue::stats->new;
+    # build hashref
+    my %queue = ();
+    foreach my $item (@{ $stats->queue_counts }) {
+        $queue{$item->{name}} = { count => $item->{count}, outstanding => [], expired => [] };
+    }
+    foreach my $item (@{ $stats->counters }) {
+        push $queue{$item->{queue}}{outstanding}, { name => $item->{name}, count => $item->{count} };
+    }
+    foreach my $item (@{ $stats->miss_counters }) {
+        push $queue{$item->{queue}}{expired}, { name => $item->{name}, count => $item->{count} };
+    }
+    #build json
+    my $json = [];
+    while( my( $key, $value ) = each %queue ){
+        # rebuild hash,
+        #   replace empty []'s w/ undef
+        push $json, { queue => $key, stats => {
+            count => $value->{count},
+            expired => ( $value->{expired}->[0] ? $value->{expired} : undef ),
```

Comment:
Would be more obvious to check if there are elements in it by doing:
```
expired => ( @{$value->{expired}} ? $value->{expired} : undef ),
```

Reply:
I'll defer this change based on your earlier comment: https://github.com/inverse-inc/packetfence/pull/3158#discussion_r183021216